src/features/student/declaredPrograms/components/interactions/formFields/DeclaredProgramPeriodFormField/DeclaredProgramPeriodFormField.test.ts

```diff
@@ -0,0 +1,253 @@
+import { AvCheckboxStub, AvInputStub, BddTest } from '@avenirs-esr/avenirs-dsav/test-utils'
+import { useForm } from '@tanstack/vue-form'
+import { mount, type VueWrapper } from '@vue/test-utils'
+import { beforeEach, expect, vi } from 'vitest'
+
+const TestWrapper = defineComponent({
+  components: {
+    DeclaredProgramPeriodFormField
+  },
+  setup () {
+    const form = useForm({
+      defaultValues: {
+        startDate: '',
+        endDate: '',
+        isOngoing: false
+      } as DeclaredProgramFormData,
+      validators: {
+        onSubmit ({ value }) {
+          const errors: Record<string, string | undefined> = {
+            startDate: undefined,
+            endDate: undefined,
+            isOngoing: undefined
+          }
+
+          if (!value.startDate || value.startDate.trim() === '') {
+            errors.startDate = 'La date de début est requise'
+          }
+
+          if (!value.isOngoing && (!value.endDate || value.endDate.trim() === '')) {
+            errors.endDate = 'La date de fin est requise'
+          }
+
+          return { fields: errors }
+        }
+      }
+    }) as unknown as AddDeclaredProgramForm
+
+    return { form }
+  },
+  template: `
+    <form @submit.prevent="form.handleSubmit">
+      <DeclaredProgramPeriodFormField :form="form" />
+    </form>
+  `
+})
+
+BddTest().given('a declared program period form field', () => {
+  let wrapper: VueWrapper<InstanceType<typeof TestWrapper>>
+
+  const stubs = {
+    AvCheckbox: AvCheckboxStub,
+    AvInput: AvInputStub
+  }
+
+  BddTest().when('the component is mounted', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(TestWrapper, {
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should render the checkbox and date inputs', () => {
+      const checkbox = wrapper.findAllComponents({ name: 'AvCheckbox' })
+      const inputs = wrapper.findAllComponents({ name: 'AvInput' })
+      expect(checkbox.length).toBe(1)
+      expect(inputs.length).toBe(2)
+    })
+
+    BddTest().then('it should have empty initial values', () => {
+      const inputs = wrapper.findAllComponents({ name: 'AvInput' })
+      expect(inputs[0].props('modelValue')).toBe('')
+      expect(inputs[1].props('modelValue')).toBe('')
+    })
+
+    BddTest().then('it should have isOngoing unchecked', () => {
+      const checkbox = wrapper.findComponent({ name: 'AvCheckbox' })
+      expect(checkbox.props('modelValue')).toEqual([])
+    })
+
+    BddTest().then('it should enable end date input', () => {
+      const inputs = wrapper.findAllComponents({ name: 'AvInput' })
+      expect(inputs[1].props('disabled')).toBe(false)
+    })
+  })
+
+  BddTest().and('the user enters a start date', () => {
+    beforeEach(async () => {
+      vi.clearAllMocks()
+      wrapper = mount(TestWrapper, {
+        global: { stubs }
+      })
```

Comment:
```suggestion
  BddTest().and('the user enters a start date', () => {
```

idem que les autres fichiers de test : and dans when et suppression du code répété, valable pour tous les and